packages/ic-signature-verification/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+use candid::Principal;
+use ic_cbor::CertificateToCbor;
+use ic_certification::{
+    leaf, Certificate, Delegation, HashTree, LookupResult, SubtreeLookupResult,
+};
+use ic_verify_bls_signature::verify_bls_signature;
+use serde::{Deserialize, Serialize};
+use serde_bytes::ByteBuf;
+use sha2::{Digest, Sha256};
+
+pub const IC_STATE_ROOT_DOMAIN_SEPARATOR: &[u8; 14] = b"\x0Dic-state-root";
+
+use ic_canister_sig_creation::{extract_raw_root_pk_from_der, CanisterSigPublicKey};
+
+/// Verifies that `signature` is a valid canister signature on `message`.
+/// https://internetcomputer.org/docs/current/references/ic-interface-spec#canister-signatures
+pub fn verify_canister_sig(
+    message: &[u8],
+    signature_cbor: &[u8],
+    public_key_der: &[u8],
```

Comment:
General note: later we'll want to use this function also directly in the replica code. There, however, the DER is already stripped away from the public key. I guess we don't have to solve this right now, but eventually we need a way to deal with this.

Another relevant aspect for using this code in the replica is _caching_: currently the replica does caching on the verified BLS signatures. Eventually we want to have that here as well, at least optionally. Most likely, the way to achieve this is to add caching capabilities to `ic_verify_bls_signature`.